### PR TITLE
ENT-4494 Harmonize network stack

### PR DESCRIPTION
**Pull request by blsemo**
Remove differences between OS and Enterprise for our amqp client and server, protonwrapper and artemis server/client implementation.

### CODE REVIEW

**Comment by vkolomeyko** [comment]
Looks fine to me in principle. I have not checked every line in every file, but I would expect code to be copied from ENT to OS.
CI builds are failing atm due to Detekt baseline not being up-to-date.

**Review by vkolomeyko — APPROVED** [review]

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/bridging/BridgeControlMessages.kt [hunk 1]
```diff
@@ -11,7 +11,7 @@ import net.corda.core.utilities.NetworkHostAndPort
  * @property legalNames The list of acceptable [CordaX500Name] names that should be presented as subject of the validated peer TLS certificate.
  */
 @CordaSerializable
-data class BridgeEntry(val queueName: String, val targets: List<NetworkHostAndPort>, val legalNames: List<CordaX500Name>)
+data class BridgeEntry(val queueName: String, val targets: List<NetworkHostAndPort>, val legalNames: List<CordaX500Name>, val serviceAddress: Boolean)
```
Comment (mnesbit) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/bridging/BridgeControlMessages.kt:14]: check in p2pmessagingclient

**Comment by roastario** [comment]
run tests: smoke

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/bridging/AMQPBridgeManager.kt [hunk 2]
```diff
@@ -119,55 +216,196 @@ class AMQPBridgeManager(config: MutualSslConfiguration,
...
+        private fun createSessionAndConsumer(startedArtemis: ArtemisMessagingClient.Started): ClientSession {
+            logInfoWithMDC("Creating session and consumer.")
+            val sessionFactory = startedArtemis.sessionFactory
+            val session = sessionFactory.createSession(NODE_P2P_USER, NODE_P2P_USER, false, true,
+                                        true, false, DEFAULT_ACK_BATCH_SIZE)
+            this.session = session
+            // Several producers (in the case of shared bridge) can put messages in the same outbound p2p queue.
+            // The consumers are created using the source x500 name as a filter
+            val consumer = if (amqpConfig.enableSNI) {
+                session.createConsumer(queueName, "hyphenated_props:sender-subject-name = '${amqpConfig.sourceX500Name}'")
```
Comment (mnesbit) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/bridging/AMQPBridgeManager.kt:335]: This is the bit that mandates a sender-subject-name is attached at node sending side as it filters by source.

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/bridging/BridgeControlMessages.kt [hunk 3]
```diff
@@ -47,4 +47,13 @@ sealed class BridgeControl {
...
     data class Delete(val nodeIdentity: String, val bridgeInfo: BridgeEntry) : BridgeControl()
+
+    /**
+     * This message is sent to Bridge to check the health of it.
+     * @property requestId The identifier for the health check request as health check is likely to be produced repeatedly.
+     * @property command  Allows to specify the sort fo health check that needs to be performed.
+     * @property bridgeInfo The connection details of the new bridge (optional).
+     */
+    @CordaSerializable
+    data class BridgeHealthCheck(val requestId: Long, val command: String, val bridgeInfo: BridgeEntry?) : BridgeControl()
```
Comment (mnesbit) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/bridging/BridgeControlMessages.kt:58]: Lets make sure all relevant when statements can safely ignore/print warning for this message

**Comment by blsemo** [comment]
I have run the bank-of-corda-demo with this branch and a `CashIssueAndPay` flow completed successfully. I checked the logs on both sides and they happily recorded txs, so I believe the the P2P communication works.

**Review by mnesbit — APPROVED** [review]
Yay!

**Pull request merged**